tests/test_analysis.py

```diff
@@ -0,0 +1,98 @@
+import covasim as cv
+import unittest
+import numpy as np
+
+
+# Suite of tests to test basic functionality of the analysis.py file
+# Runtime: 5.004 seconds
+def test_analysis_snapshot():
+    sim = cv.Sim(analyzers=cv.snapshot('2020-04-04', '2020-04-14'))
+    sim.run()
+    snapshot = sim['analyzers'][0]
+    people1 = snapshot.snapshots[0]            # Option 1
+    people2 = snapshot.snapshots['2020-04-04'] # Option 2
+    people3 = snapshot.get('2020-04-14')       # Option 3
+    people4 = snapshot.get(34)                 # Option 4
+    people5 = snapshot.get()                   # Option 5
+    # people3 = []  # uncomment to verify error
+    peoples = [people1, people2, people3, people4, people5]
+    for i, people in enumerate(peoples):
+        optionNum = i+1
+        assert len(people) > 0, f"Option {optionNum} should have more than 0 members"
+
+
+
+def test_analysis_hist():
+    # raising multiple histograms to check windows functionality
+    age_analyzer = cv.age_histogram(days=["2020-03-30", "2020-03-31", "2020-04-01"])
+    sim = cv.Sim(analyzers=age_analyzer)
+    sim.run()
+
+    # checks to make sure dictionary form has right keys
+    agehistDict = sim['analyzers'][0].get()
+    assert len(agehistDict.keys()) == 5
+
+    # checks to see that compute windows is correct
+    agehist = sim['analyzers'][0]
+    try:
+        agehist.compute_windows()
+    except ValueError:
+        raise ValueError("Unable to compute windows")
+
+    # checks compute_windows and plot()
+    try:
+        agehist.plot(windows=True)
+    except:
+        raise ValueError("Cannot plot this histogram with windows")
+
+
+def test_analysis_fit():
+    sim = cv.Sim(datafile="example_data.csv")
+    sim.run()
+    fit = sim.compute_fit()

```

Comment:
Might want to find out what this object is here and assert on its properties, then check these assertions again after all of the manipulations.

Reply:
Okay, I am going to check the results of these functions before and after adding custom data to ensure that internal state is changing appropriately.